File: 251004/Kurilovich/labs/main_app/app/Models/Topic.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Topic extends Model
{
    public $table = 'tbl_topic';

    public $fillable = [
        'editor_id',
        'title',
        'content',
    ];

    public function tags() {
        return $this->belongsToMany(Tag::class, 'tbl_topic_tag', 'topic_id', 'tag_id');
    }
}

File: 251001/Churavskiy/lab3/publisher/src/main/java/com/example/publisher/entity/Topic.java
package com.example.publisher.entity;

import com.example.publisher.entity.BaseEntity;
import com.example.publisher.entity.Tag;
import com.example.publisher.entity.User;
import jakarta.validation.constraints.NotNull;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Set;

@Entity
@Table(name = "tbl_topic", uniqueConstraints = {
        @UniqueConstraint(columnNames = "title")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Topic implements BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String title;

    @Column(length = 2048)
    private String content;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private LocalDateTime created;

    private LocalDateTime modified;

    @ManyToMany(cascade = CascadeType.REMOVE)
    @JoinTable(
            name = "tbl_topic_tag",
            joinColumns = @JoinColumn(name = "topic_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_id")
    )
    private Set<Tag> tags;
}
